pkg/ui/workspaces/crdb-js-proto/BUILD.bazel

```diff
@@ -0,0 +1,82 @@
+load("@aspect_rules_js//js:defs.bzl", "js_binary", "js_run_binary", "js_test")
+load("@aspect_rules_js//npm:defs.bzl", "npm_package")
+load("@aspect_rules_ts//ts:defs.bzl", "ts_project")
+load("@bazel_skylib//rules:write_file.bzl", "write_file")
+load("@npm//:defs.bzl", "npm_link_all_packages")
+load("@npm//:defs.bzl", "npm_link_all_packages")
+load("//docs/generated/http:defs.bzl", "PROTOBUF_TARGETS")
+load("//pkg/ui/workspaces/db-console/src/js:defs.bzl", "proto_sources")
+
+npm_link_all_packages(name = "node_modules")
+
+genrule(
+    name = "protos",
+    outs = ["dist"],
+    cmd = """ \
+    mkdir -p $(OUTS); \
+    DESCRIPTOR_SET_IN=$$(for word in $(SRCS); do echo $$word; done | grep '\\.bin$$' | tr -s '\\n' ':'); \
+    PROTO_PATHS=$$(for f in $(locations :proto_sources); \
+        do if [[ "$$f" =~ ^.*/pkg/.*/_virtual_imports/.*_proto/.*.proto$$ ]]; \
+            then echo $$f | sed 's|.*pkg/.*/_virtual_imports/.*_proto/||' | sed 's|.bin$$||'; \
+        fi; done | sort -u); \
+    export PATH=$$PATH:$$(dirname $(NODE_PATH)); \
+    $(location @com_google_protobuf//:protoc) \
```

Comment:
For cases like this, some indentation is useful to distinguish where the command begins and ends (and the backslashes are indeed mandatory because you're splitting one command across many lines)